omnimove_external_control/src/external_control_message.cpp

```diff
@@ -0,0 +1,194 @@
+#include <arpa/inet.h>
+#include <rcl/logging.h>
+#include <boost/asio.hpp>
+#include <cstring>
+#include <omnimove/external_control_message.hpp>
+
+namespace omnimove
+{
+
+ExternalControlMessage::ExternalControlMessage(
+  const char * header, const char * message, const int & data_length)
+{
+  size_t header_length = strlen(header);
+  msg_length_ = header_length + data_length + 2;
+  msg_buffer_ = std::make_unique<char[]>(msg_length_);
```

Comment:
this will be only a problem, when going for real-time, but any allocation on the heap is not allowed in a real-time context (heap allocations should happen in the initialization phase, not in the cyclic part, this constructor is called in every cycle)